bin/fixmystreet.com/bulk-report-close

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env perl
+
+# bulk close a list of reports, with a message and a state.
+# uses a CSV file with a list of report ids and a text file with the message
+
+use utf8;
+use v5.14;
+use warnings;
+
+BEGIN {
+    use File::Basename qw(dirname);
+    use File::Spec;
+    my $d = dirname(File::Spec->rel2abs($0));
+    require "$d/../../setenv.pl";
+}
+
+use Path::Tiny;
+use Text::CSV;
+use FixMyStreet::DB;
+use FixMyStreet::Script::ArchiveOldEnquiries;
+use Getopt::Long::Descriptive;
+
+my ($opts, $usage) = describe_options(
+    '%c %o',
+    ['reports=s', "csv file with a single column of report ids", { required => 1 } ],
+    ['message=s', "text file with the message to add to reports", { required => 1 } ],
+    ['state=s', "state to set reports to", { required => 1 } ],
+    ['body=s', "body to use for the comments", { required => 1 } ],
+    ['commit|c', "actually close reports and add comments. Omitting this flag will do a dry-run"],
+    ['help|h', "print usage message and exit" ],
+);
+print($usage->text), exit if $opts->help;
+
+my $body = FixMyStreet::DB->resultset("Body")->search({ name => $opts->{body} })->first;
+die "Could not find body '$opts->{body}'\n" unless $body;
+
+my $file = path($opts->{message});
+my $message = $file->slurp;
+
+die "No message provided\n" unless $message;
+
+my $state = FixMyStreet::DB->resultset("State")->search({ label => $opts->{state} })->first;
+die "Could not find state '$opts->{state}'\n" unless $state;
+
+my @report_ids;
+
+my $csv = Text::CSV->new;
+open my $fh, "<:encoding(utf-8)", $opts->{reports} or die "Failed to open $file: $!\n";
+while (my $line = $csv->getline($fh)) {
+    push @report_ids, $line->[0] if $line->[0] =~ m/^\d*$/;
+}
+
+my $email_cutoff = DateTime->now->add('months', -6)->truncate( to => 'day' );
+
+my $rs  = FixMyStreet::DB->resultset("Problem")->search({
+    id => \@report_ids,
+    state => [ FixMyStreet::DB::Result::Problem->open_states() ],
+})->to_body( $body );
+
+my $no_message = $rs->search({
+    lastupdate => { '<', $email_cutoff->ymd },
+});
+
+my $with_message = $rs->search({
+    lastupdate => { '>=', $email_cutoff->ymd },
+});;
+
+FixMyStreet::Script::ArchiveOldEnquiries::update_options({
+    user => $body->comment_user->id,
+    closure_text => $message,
+    retain_alerts => 0,
+    commit => $opts->commit,
+    closed_state => $opts->{state},
+});
+
+if ($opts->{commit}) {
+    if ($no_message->count > scalar(@report_ids)) {
+        die "Found more reports than expected, bailing";
+    }
+    printf("Closing %d old reports with no alert ", $no_message->count);
+    FixMyStreet::Script::ArchiveOldEnquiries::close_problems($no_message);
+    printf("done.\n");
+} else {
+    printf("Would close %d old reports with no alert. Run with --commit to actually close reports.\n", $no_message->count);
+}
+
+# Provide some variables to the archiving script
```

Comment:
This doesn't change any of the options, should it be called with retain_alerts = 1?